src/file_op.rs

```diff
@@ -21,11 +21,34 @@ use crate::{
 /// Returns total bytes written to file.
 async fn save_file(
     field: actix_multipart::Field,
-    file_path: PathBuf,
+    mut file_path: PathBuf,
     overwrite_files: bool,
+    rename_duplicate: bool,
 ) -> Result<u64, RuntimeError> {
-    if !overwrite_files && file_path.exists() {
-        return Err(RuntimeError::DuplicateFileError);
+    if file_path.exists() {
+        // clap makes sure both overwrite_files and rename_duplicate cannot be true
+        if rename_duplicate {
+            // optionally should we use Path::file_prefix, which
+            // extracts the portion of the file name before the
+            // first .?
+            let file_name = file_path.file_stem().unwrap_or_default().to_string_lossy();
+            let file_ext = file_path.extension().map(|s| s.to_string_lossy());
+            let filepaths = (1..).map(|i| {
+                if let Some(ext) = &file_ext {
+                    file_path.with_file_name(format!("{}-{}.{}", file_name, i, ext))
+                } else {
+                    file_path.with_file_name(format!("{}-{}", file_name, i))
+                }
+            });
+            for fp in filepaths {
+                if !fp.exists() {
+                    file_path = fp;
+                    break;
+                }
+            }
```

Comment:
This is a little bit dense and not super self-descriptive. Could you put in some comments describing the idea and/or rewrite the code a bit to be more self-descriptive?

Also, shouldn't always just go `.n` perhaps? I'm not really sure how other systems do this but maybe we can follow some convention that users could have come to expect from elsewhere?

Reply:
It doesn't go .n, it adds the number after the filename and before the extension. It's how Firefox download works. 
`file.txt` becomes `file-1.txt` if it already exists (or `file-2.txt` and so on), the dense and complicated looking bits is because rust needs complicated logic to extract file-stem and extension... I'll see if I can extract this logic to a function and explain it there. 

The core logic is:
get file_stem (without extension)
get file_extension 

Loop for N from 1 to ∞:
- if file has extension: do file_stem-N.extension
- if file has no extension: do file_stem-N
- check if new name already exists: if not, exit else increment N

Reply:
I know what the code does, I just found it a little hard to read and would like some code comments at the very least to make sure the desired logic is abundantly clear. :)

Reply:
I made a for loop with the numbers instead, it should be easier to read. Also added comments.